appservice/src/deploy/deploy.ts

```diff
@@ -3,121 +3,24 @@
-        if (context.stopAppBeforeDeploy) {
-            ext.outputChannel.appendLog(l10n.t('Stopping app...'), { resourceName: site.fullName });
-            await client.stop();
-        }
-
-        ext.outputChannel.appendLog(l10n.t('Starting deployment...'), { resourceName: site.fullName });
-        try {
-            if (!context.deployMethod && config.scmType === ScmType.GitHub) {
-                throw new Error(l10n.t('"{0}" is connected to a GitHub repository. Push to GitHub repository to deploy.', site.fullName));
-            } else if (!context.deployMethod && config.scmType === ScmType.LocalGit) {
-                await localGitDeploy(site, { fsPath: fsPath }, context);
-            } else {
-                if (!(await fse.pathExists(fsPath))) {
-                    throw new Error(l10n.t('Failed to deploy path that does not exist: {0}', fsPath));
-                }
-
-                const javaRuntime = site.isLinux ? config.linuxFxVersion : config.javaContainer;
-                if (javaRuntime && /^(tomcat|wildfly|jboss)/i.test(javaRuntime)) {
-                    await deployWar(context, site, fsPath);
-                } else if (javaRuntime && /^java/i.test(javaRuntime) && !site.isFunctionApp) {
-                    const pathFileMap = new Map<string, string>([
-                        [path.basename(fsPath), 'app.jar']
-                    ]);
-                    await deployZip(context, site, fsPath, aspPromise, pathFileMap);
-                } else if (context.deployMethod === 'storage') {
-                    await deployToStorageAccount(context, fsPath, site);
-                } else {
-                    await deployZip(context, site, fsPath, aspPromise);
-                }
-            }
-        } finally {
-            if (context.stopAppBeforeDeploy) {
-                ext.outputChannel.appendLog(l10n.t('Starting app...'), { resourceName: site.fullName });
-                await client.start();
-            }
-        }
-
-        await startPostDeployTask(context, fsPath, config.scmType, site.fullName);
-
-        if (context.syncTriggersPostDeploy) {
-            // Don't sync triggers if app is stopped https://github.com/microsoft/vscode-azurefunctions/issues/1608
-            const state: string | undefined = await client.getState();
-            if (state?.toLowerCase() === 'running') {
-                await syncTriggersPostDeploy(context, site);
-            }
-        }
-    });
-}
-
-function getLinuxFxVersionForTelemetry(config: SiteConfigResource): string {
-    const linuxFxVersion = config.linuxFxVersion || '';
-    // Docker values point to the user's specific image, which we don't want to track
-    return /^docker/i.test(linuxFxVersion) ? 'docker' : linuxFxVersion;
+    const innerContext: InnerDeployContext = Object.assign(context, { site, fsPath, client, aspPromise });
+    const title: string = l10n.t('Deploying to function app "{0}"', site.fullName);
```

Comment:
Function is used by App Service and Functions so title needs to either be generic or handle both cases.

```suggestion
    const title: string = l10n.t('Deploying to "{0}"', site.fullName);
```